scripts/imgtool/boot_record.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2019, Arm Limited.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import sys
+import cbor
+
+
+# SW component IDs
+SW_COMPONENT_RANGE = 0
+SW_COMPONENT_TYPE = SW_COMPONENT_RANGE + 1
+MEASUREMENT_VALUE = SW_COMPONENT_RANGE + 2
+SW_COMPONENT_VERSION = SW_COMPONENT_RANGE + 4
+SIGNER_ID = SW_COMPONENT_RANGE + 5
+MEASUREMENT_DESCRIPTION = SW_COMPONENT_RANGE + 6
```

Comment:
Why not use an `Enum`?

Reply:
I'm not sure if it's worth it. For example the `SIGNER_ID` as an enum would look like: `SwCompFields.SIGNER_ID.value`.

Reply:
> I'm not sure if it's worth it. For example the `SIGNER_ID` as an enum would look like: `SwCompFields.SIGNER_ID.value`.

A better naming would look like this: `SwComponent.SIGNER_ID`

Reply:
Btw, why is `SW_COMPONENT_VERSION = SW_COMPONENT_RANGE + 4` using `+ 4` instead of `+ 3`?

Reply:
These SW component property IDs are specified by PSA (in the PSA Attestation API document).
The ID `3` is Reserved.